packages/x-charts/src/ChartsVoronoiHandler/ChartsVoronoiHandler.tsx

```diff
@@ -70,7 +71,17 @@ function ChartsVoronoiHandler(props: ChartsVoronoiHandlerProps) {
       const getXPosition = getValueToPositionMapper(xScale);
       const getYPosition = getValueToPositionMapper(yScale);
 
-      const seriesPoints = data.flatMap(({ x, y }) => [getXPosition(x), getYPosition(y)]);
+      const seriesPoints = data.flatMap(({ x, y }) => {
+        const pointX = getXPosition(x);
+        const pointY = getYPosition(y);
+
+        if (!drawingArea.isPointInside({ x: pointX, y: pointY })) {
+          return [-drawingArea.width, -drawingArea.height];
+        }
```

Comment:
Just to be kind with our future self in 1 year asking why such values :)

```suggestion
        if (!drawingArea.isPointInside({ x: pointX, y: pointY })) {
          // If the point is not displayed we move them to a trash coordinate.
          // This avoids managing index mapping before/after filtering.
          // The trash point is far enough such that any point in the drawing area will be closer to the mouse than the trash coordinate.
          return [-drawingArea.width, -drawingArea.height];
        }
```